view/stache/mustache_helpers.js

```diff
@@ -150,10 +150,11 @@ steal("can/util", "./utils.js","can/view/live",function(can, utils, live){
 			};
 		},
 		'switch': function(expression, options){
+			var expressionValue = resolve(expression);
 			var found = false;
 			var newOptions = options.helpers.add({
 				case: function(value, options){
-					if(resolve(expression) === resolve(value)) {
+					if(!found && expressionValue === resolve(value)) {
```

Comment:
I don't know, I did because it seemed like if a case is already found at this point there is no reason to evaluate the value here. We can still talk about it though.